apps/sms/views/inbox/js/inbox.js

```diff
@@ -501,17 +501,56 @@ var InboxView = {
       });
     }
 
-    return Utils.confirm(
-      {
-        id: 'deleteThreads-confirmation-message',
-        args: { n: selected.length }
-      },
-      null,
-      {
-        text: 'delete',
-        className: 'danger'
+    var unreadMessagesCount = 0; 
+    var unreadThreadIds = [];
+
+    // Check if all threads are read.
+    unreadThreadIds = selected.reduce(function(list, threadId) {
+      if(Threads.get(threadId).unreadCount) {
+        unreadMessagesCount += Threads.get(threadId).unreadCount;
+        list.push(threadId);
       }
-    ).then(performDeletion.bind(this));
+
+      return list;
+    }, []);
+    
+    if(unreadThreadIds.length) {
+      Utils.confirm(
+        {
+          id: 'messages',
```

Comment:
@julienw , Can we set two l10n-args in one l10n string?

regarding this "The thread(s) you selected contains unread message(s). Delete anyway? ".